app/Repositories/Promotion/PromotionRepository.php

```diff
@@ -19,4 +20,18 @@ public function __construct(Promotion $model)
     {
         $this->model = $model;
     }
+    /**
+     * [getByProduct description]
+     *
+     * @param [type] $search [Search product by promotion]
+     * @param [type] $limit  [limit colum]
+     *
+     * @return [type]         [get all colum of search]
+     */
+    public function getByPromotion($search, $limit)
+    {
+        return $this->model->with('product')->whereHas('product', function ($query) use ($search) {
+            $query->where('product_name', 'like', '%'.$search.'%')->where('date_start', '<=', date('Y-m-d H:i:s'))->where('date_end', '>=', date('Y-m-d H:i:s'));
```

Comment:
[rank C] do you know datetime format is defining in config files? please avoid to hardcode it.